diff --git a/vendor/mediatek/kernel_modules/mtkcam/mtk-mae/isp8/mtk-mae-isp8.c b/vendor/mediatek/kernel_modules/mtkcam/mtk-mae/isp8/mtk-mae-isp8.c
index 1dc763927..732900ada 100644
--- a/vendor/mediatek/kernel_modules/mtkcam/mtk-mae/isp8/mtk-mae-isp8.c
+++ b/vendor/mediatek/kernel_modules/mtkcam/mtk-mae/isp8/mtk-mae-isp8.c
@@ -7,6 +7,7 @@
  */
 
 #include <linux/module.h>
+#include <linux/minmax.h>
 #include <linux/platform_device.h>
 
 #include <linux/soc/mediatek/mtk-cmdq-ext.h>
@@ -44,8 +45,12 @@
 // #define FLOOR_NEG(X)		((X-(int)(X)) < 0 ? (int)(X-1) : (int)(X))
 // #define FLOOR(X)		( ((X) > 0) ? FLOOR_POS(X) : FLOOR_NEG(X) )
 
+#ifndef MIN
 #define MIN(X,Y)		(((X) > (Y)) ? (Y) : (X))
+#endif
+#ifndef MAX
 #define MAX(X,Y)		(((X) > (Y)) ? (X) : (Y))
+#endif
 #define ABS(X)			(((X) > 0) ? (X) : -(X))
 #define ROUND(X,Y)		((int)((2*(X))+(Y))/(2*(Y)))
 #define BUFFER_NAME_LEN		50
