diff --git a/vendor/mediatek/kernel_modules/gpu/gpu_mali/mali_avalon/mali-r49p1/drivers/gpu/arm/midgard/mali_kbase_js.c b/vendor/mediatek/kernel_modules/gpu/gpu_mali/mali_avalon/mali-r49p1/drivers/gpu/arm/midgard/mali_kbase_js.c
index 07042508f..66d7810b6 100644
--- a/vendor/mediatek/kernel_modules/gpu/gpu_mali/mali_avalon/mali-r49p1/drivers/gpu/arm/midgard/mali_kbase_js.c
+++ b/vendor/mediatek/kernel_modules/gpu/gpu_mali/mali_avalon/mali-r49p1/drivers/gpu/arm/midgard/mali_kbase_js.c
@@ -23,6 +23,7 @@
  * Job Scheduler Implementation
  */
 #include <mali_kbase.h>
+#include <linux/cred.h>
 #include <mali_kbase_js.h>
 #include <tl/mali_kbase_tracepoints.h>
 #include <mali_linux_trace.h>
@@ -155,10 +156,10 @@ static inline int gpu_metrics_ctx_init(struct kbase_context *kctx)
 	unsigned long flags;
 	int ret = 0;
 
-	const struct cred *cred = get_current_cred_module();
+	const struct cred *cred = get_current_cred();
 	const unsigned int aid = cred->euid.val;
 
-	put_cred_module(cred);
+	put_cred(cred);
 
 	/* Return early if this is not a Userspace created context */
 	if (unlikely(!kctx->filp))
@@ -3332,4 +3333,4 @@ base_jd_prio kbase_js_priority_check(struct kbase_device *kbdev, base_jd_prio pr
 		out_priority = pcm_device->ops.pcm_scheduler_priority_check(pcm_device, current,
 									    req_priority);
 	return kbasep_js_sched_prio_to_atom_prio(kbdev, out_priority);
-}
+}
\ No newline at end of file
